src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/TagBuilder.cs

```diff
@@ -25,15 +27,25 @@ public TagBuilder(string tagName)
             }
 
             TagName = tagName;
-            Attributes = new SortedDictionary<string, string>(StringComparer.OrdinalIgnoreCase);
-
             InnerHtml = new BufferedHtmlContent();
         }
 
         /// <summary>
         /// Gets the set of attributes that will be written to the tag.
-        /// </summary>
-        public IDictionary<string, string> Attributes { get; }
+        /// </summary>B
+        public AttributeDictionary Attributes
```

Comment:
If you guys are really opposed to the idea of exposing the concrete type then the right thing to do is to start over with this API altogether.